src/main/java/frc/trigon/robot/subsystems/elevator/ElevatorConstants.java

```diff
@@ -1,50 +1,157 @@
+    static final boolean FOC_ENABLED = true;
+    static final double
             DRUM_RADIUS_METERS = 0.0222997564,
-            DRUM_DIAMETER_METERS = DRUM_RADIUS_METERS * 2;
-    public static final double GEAR_RATIO = 3.44;
-    static final double CAMERA_PLATE_HEIGHT_METERS = 0.190193;
-    static final double TOLERANCE_METERS = 0.035;
+            DRUM_DIAMETER_METERS = DRUM_RADIUS_METERS * 2,
+            MAXIMUM_HEIGHT_METERS = 1.111,
+            CAMERA_PLATE_HEIGHT_METERS = 0.190193,
+            RETRACTED_ELEVATOR_LENGTH_METERS = 0.63;
+
     static final SysIdRoutine.Config SYSID_CONFIG = new SysIdRoutine.Config(
             Units.Volts.of(0.25).per(Units.Second.of(1)),
             Units.Volts.of(2),
             Units.Second.of(1000)
     );
+
     static final Pose3d
             ELEVATOR_ORIGIN_POINT = new Pose3d(0.10018, 0, 0.04, new Rotation3d(0, edu.wpi.first.math.util.Units.degreesToRadians(10), 0)),
             TRANSPORTER_ORIGIN_POINT = new Pose3d(0.10018, 0, 0.06, new Rotation3d(0, edu.wpi.first.math.util.Units.degreesToRadians(10), 0));
-    private static final double
-            ELEVATOR_MECHANISM_WIDTH = 3,
-            ELEVATOR_MECHANISM_HEIGHT = 3,
-            ELEVATOR_MECHANISM_ROOT_X = 1.44,
-            ELEVATOR_MECHANISM_ROOT_Y = 0,
-            TARGET_ELEVATOR_POSITION_MECHANISM_ROOT_X = 1.57,
-            TARGET_ELEVATOR_POSITION_MECHANISM_ROOT_Y = 0,
-            LIGAMENT_LINE_WIDTH = 10;
-    static final Mechanism2d ELEVATOR_MECHANISM = new Mechanism2d(
-            ELEVATOR_MECHANISM_WIDTH,
-            ELEVATOR_MECHANISM_HEIGHT
+    static final ElevatorMechanism2d MECHANISM = new ElevatorMechanism2d(
+            "Elevator",
+            MAXIMUM_HEIGHT_METERS,
+            RETRACTED_ELEVATOR_LENGTH_METERS,
+            new Color8Bit(Color.kYellow)
     );
-    private static final MechanismRoot2d
-            ELEVATOR_ROOT = ELEVATOR_MECHANISM.getRoot("ElevatorRoot", ELEVATOR_MECHANISM_ROOT_X, ELEVATOR_MECHANISM_ROOT_Y),
-            TARGET_ELEVATOR_POSITION_ROOT = ELEVATOR_MECHANISM.getRoot("TargetElevatorPositionRoot", TARGET_ELEVATOR_POSITION_MECHANISM_ROOT_X, TARGET_ELEVATOR_POSITION_MECHANISM_ROOT_Y);
-    private static final Color8Bit MECHANISM_COLOR = new Color8Bit(Color.kYellow);
-    static final MechanismLigament2d
-            ELEVATOR_LIGAMENT = ELEVATOR_ROOT.append(new MechanismLigament2d("ElevatorLigament", 0, 90, LIGAMENT_LINE_WIDTH, MECHANISM_COLOR)),
-            TARGET_ELEVATOR_POSITION_LIGAMENT = TARGET_ELEVATOR_POSITION_ROOT.append(new MechanismLigament2d("TargetElevatorPositionLigament", 0, 90, LIGAMENT_LINE_WIDTH, new Color8Bit(Color.kGray)));
+    private static final int MOTOR_AMOUNT = 2;
+    private static final DCMotor GEARBOX = DCMotor.getKrakenX60Foc(MOTOR_AMOUNT);
+    private static final ElevatorSimulation SIMULATION = new ElevatorSimulation(GEARBOX, GEAR_RATIO, 1, DRUM_RADIUS_METERS, RETRACTED_ELEVATOR_LENGTH_METERS, MAXIMUM_HEIGHT_METERS, true);
```

Comment:
the sim stuff should be declared right after motor settings
why is the mass a magic number (1)?

Reply:
What should I put there?

Reply:
Create a constant of the mass, and use the mass that was declared in the (SimulationElevatorConstants)[https://github.com/Programming-TRIGON/RobotCode2024/blob/ae53b8c8530db3d75c0420dc652884022602bea6/src/main/java/frc/trigon/robot/subsystems/elevator/simulationelevator/SimulationElevatorConstants.java#L11]

Reply:
Like what should the mass be?

Reply:
the mass declared in the sim constants beforehand

Reply:
I can't find it